torch/testing/_internal/common_methods_invocations.py

```diff
@@ -408,7 +410,10 @@ def sample_inputs(self, device, dtype, requires_grad=False):
                    )),
     UnaryUfuncInfo('sinh',
                    ref=np.sinh,
```

Comment:
Just make the reference a function that wraps sinh and promotes integer types to float before calling it. Alternatively it can use NumPy sinh's `dtype` kwarg to enforce the computation occurs in the same dtype as in PyTorch. That way the logic can live in the reference.